tools/label-check.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import glob
+import sys
+
+
+LOCALS_TEMPLATE = '''
+locals {{
+  # This label allows for billing report tracking based on module.
+  labels = merge(var.labels, {{ ghpc_module = \"{module_label}\" }})
+}}
+'''
+
+
+class ModulePath:
+    """Convenience class to get various path related information about a module"""
+    MAIN_FILENAME = "main.tf"
+    VARS_FILENAME = "variables.tf"
+    VERSIONS_FILENAME = "versions.tf"
+    OUTPUT_FILENAME = "outputs.tf"
+
+    def __init__(self, module_path: str):
+        self.module_path = module_path
+
+    def has_main(self) -> str:
+        return os.path.isfile(os.path.join(self.module_path, self.MAIN_FILENAME))
+
+    def has_vars(self) -> str:
+        return os.path.isfile(os.path.join(self.module_path, self.VARS_FILENAME))
+
+    def has_outputs(self) -> str:
+        return os.path.isfile(os.path.join(self.module_path, self.OUTPUT_FILENAME))
+
+    def main(self) -> str:
+        return self._filepath(self.MAIN_FILENAME)
+
+    def vars(self) -> str:
+        return self._filepath(self.VARS_FILENAME)
+
+    def versions(self) -> str:
+        return self._filepath(self.VERSIONS_FILENAME)
+
+    def outputs(self) -> str:
+        return self._filepath(self.OUTPUT_FILENAME)
+
+    def primary_file(self) -> str:
+        """The file that should contain the labels definition"""
+        return self.main() if self.has_main() else self.outputs()
+
+    def name(self) -> str:
+        return os.path.basename(self.module_path)
+
+    def name_label(self) -> str:
+        return self.name().lower()
+
+    def _filepath(self, name: str) -> str:
+        return os.path.join(self.module_path, name)
+
+
+def get_module_paths(root_dir="./") -> list[str]:
```

Comment:
nit. What type check did you use? IIUC, some (namely Pyre) doesn't like `list` and requires `typing.List` used instead.

Reply:
1. I had not enabled type checking. Since enabled and found some additional misses.
2. Using pylance which does not seem to care about list vs List
3. Updated to use List anyway as it is correct